### DISCUSSION TITLE
[Cancelled] v1.5.0 Release and Upgrade

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#5420] [Announcement]
>:warning: 2021-01-29, 03:30 UTC: The release and upgrade is postponed due to [the conflic of CNY holiday](https://github.com/filecoin-project/lotus/discussions/5439).

> If you have any concerns or question, leave a comment [here](https://github.com/filecoin-project/lotus/discussions/5439).

**Comment by jennijuju** [comment]
## 2021-01-26, 00:00 EST - Calibration-net v1.5.0-pre1
🙌🏻The calibration-net has been updated to [v1.5.0-pre1](https://github.com/filecoin-project/lotus/releases/tag/v1.5.0-pre1), and the upgrade will be at epoch **278026** (around 07:00 UTC on Jan 27th, 2021, 26 hours from now). Anyone running a node on the network should upgrade to the [ntwk-calibration/1.5.1-pre1](https://github.com/filecoin-project/lotus/releases/tag/v1.5.0-pre1) release (or the latest `maste`r commit, which is `d5addf7`).
Simply run `make calibnet` to build Lotus and connect with calibration-net!

**Reply by jennijuju** [reply]
## Calibration-net has passed upgrade epoch

The calibration-net has passed the v1.5.0 upgrade epoch and new blocks have been received in the network. You can run `./lotus state get-actor t01000` and see if it returns  `(fil/3/storageminer)` to check if the state migration was successful. We are still monitoring the network and we will  send another update later to confirm the upgrade status.

**Reply by jennijuju** [reply]
## :warning: The Calibration-net upgrade was not 100% successful

Unfortunately, the calibration-net doesn't seem to be healthy(a little bit forky) after the upgrade and we are actively investigating it.

**Reply by jennijuju** [reply]
## Re: :warning: The Calibration-net upgrade was not 100% successful

We think we found the issue and will be working on a patch after more testing. I will send out another update with instructions when it is ready!

**Reply by jennijuju** [reply]
## Re:Re: :warning: The Calibration-net upgrade was not 100% successful

We have identified the issue and will be coordinating a patched release within 10 hours. We are still trying to figure out the best way to bring the network back to normal, as for now,  please *shut down your miners* so we don't keep growing the unhealthy chain.

**Comment by jennijuju** [comment]
## Calibration-net v1.5.0-pre2

The patch release [ v1.5.0-pre2](https://github.com/filecoin-project/lotus/releases/tag/v1.5.0-pre2) is out, and the upgrade will be at epoch **280426**(around 03:00 UTC on Jan 28th, 2021). Anyone running a node on the network should upgrade to the[1.5.0-pre2](https://github.com/filecoin-project/lotus/releases/tag/v1.5.0-pre2) release (or the latest master commit, which is `b76fae3`).
To upgrade, run:
- Build lotus and connect with calibration-net by running `make calibnet && ./lotus daemon`  
- Run `./lotus chain sethead --epoch 278026`. This is to roll back the chain and undone the failed upgrade.
- Restart your daemon.

You may notice some weird behaviors before the chain remains healthy, like some null tipsets, miner getting slashed and losing some power due to the slashing filer getting triggered  (sorry :pensive:) and so on. We will keep monitoring the network and keep you updated!

**Reply by arajasek** [reply]
## Re: :warning: The Calibration-net upgrade was not 100% successful

We found a teeny bug, that we've patched! If we have the infra resources, we'll get the network fixed in a couple hours, if not it'll happen on the 28th (Thursday).

For now, **all miners and node operators are recommended to shut down their nodes** -- mining blocks on this (wrong) fork will make it harder to produce the correct chain.

**Comment by jennijuju** [comment]
## Calibration-net v1.5.0-pre3

The patch release [ v1.5.0-pre3](https://github.com/filecoin-project/lotus/releases/tag/v1.5.0-pre3) is out, and the upgrade will be at epoch ** 282586**(around 21:00 UTC on Jan 28th, 2021). Anyone running a node on the network should upgrade to the[1.5.0-pre3](https://github.com/filecoin-project/lotus/releases/tag/v1.5.0-pre3) release (or the latest master commit, which is `7e55c44`).
To upgrade, run:
- Build lotus and connect with calibration-net by running `make calibnet && ./lotus daemon`  
- Run `./lotus chain sethead --epoch 280425 `. This is to roll back the chain and undone the failed upgrade.
- Restart your daemon.

Thank you all so much for keeping up with this calibration upgrade with us, this is definitely a difficult but exciting one! Let's start to test, again!

**Comment by jennijuju** [comment]
## R.I.P Calibration-net, fow now

Unfortunately, round 3 of calibration-net also failed. The actor upgrade itself, the v3 migration, works as we can see that the nodes that has upgraded to v1.5.0-pre3 runs `fil/3/storageminer`, however, <50% power of the network upgraded to the latest version and lead to the chain with the earlier version is the main chain. At this point, we decided that it doesn't make any sense to continue re-upgrading and we want to announce that the calibnet will be temporarily retired and waiting for a reset.

**That being said, we also want to cancel the current v1.5.0 release schedule, primarily because [many Chinese miners mentioned](https://github.com/filecoin-project/lotus/discussions/5439) that the proposed release and upgrade schedule conflicts with CNY holidays** and we want everyone to be able to spend valuable time with their family and loved ones.  Another reason is that we are not able to test the new code excessively and we don’t want to put the network in any risk. We will soon work on figuring out **the right cryptoecon parameters that can keep the testnet in a healthy state with a reasonable baseline**, then reset the calib-net and start testing again.

I want to thank the community that is running a node and/or a miner on calibration net for keeping up with us and participating in the past three upgrades, throughout days and nights. Thank you for being so supportive💙.
